src/util.c

```diff
@@ -2473,13 +2473,11 @@ static const uint8_t _limited_term_allowed_bitmap[16] = {
 natsStatus
 nats_validateLimitedTerm(const char *name, const char *term)
 {
-    int i = 0;
-    int len = (int) strlen(term);
 
-    if (len == 0)
+    if (nats_IsStringEmpty(term))
         return nats_setError(NATS_INVALID_ARG, "%s must not be be empty", name);
 
-    for (; term[i] != '\0'; ++i) {
+    for (int i=0; term[i] != '\0'; ++i) {
         if (i >= 16)
             return nats_setError(NATS_INVALID_ARG, "%s must not be longer than 16 characters", name);
 
```

Comment:
I see that @mtmk posted a comment on the `_limited_term_allowed_bitmap`, we could also simply do the checks down below and make sure `c` is valid without the use of such function.